### PR TITLE
Fix for unparseable error returned by Azure CNI

### PR DESCRIPTION
**Pull request by tamilmani1989**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #
This PR fixes the unparseable error returned by Azure CNI (#195 ). Azure CNI should write result to stdout only in successful case

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: cni/network/network.go [hunk 1]
```diff
@@ -151,32 +151,34 @@ func (plugin *netPlugin) Add(args *cniSkel.CmdArgs) error {
...
-		log.Printf("[cni-net] ADD command completed with result:%+v err:%v.", result, err)
+			// Convert result to the requested CNI version.
+			res, err := result.GetAsVersion(nwCfg.CNIVersion)
+			if err != nil {
+				err = plugin.Error(err)
+			}
+
+			// Output the result to stdout.
+			res.Print()
+			log.Printf("[cni-net] ADD command completed with result:%+v err:%v.", result, err)
```
Comment (sharmasushant) [inline cni/network/network.go]: We should still write to the log file what happened to the CNI ADD call

Comment (tamilmani1989) [reply cni/network/network.go]: ok

**Comment by PatrickLang** [comment]
How soon can we get a release once this is merged? I'd like to run this through the conformance tests to see if it improves the pass rate.

**Comment by PatrickLang** [comment]
@sharmasushant - does the change look good? Launching tests this afternoon

**Comment by PatrickLang** [comment]
The fixes look good. Error handling is better. I added example logs to #195 with the fix

**Comment by daschott** [comment]
@sharmasushant anything else needed for merge here?

File: cni/network/network.go [hunk 2]
```diff
@@ -169,13 +165,17 @@ func (plugin *netPlugin) Add(args *cniSkel.CmdArgs) error {
...
-		res, err := result.GetAsVersion(nwCfg.CNIVersion)
-		if err != nil {
-			err = plugin.Error(err)
+		res, vererr := result.GetAsVersion(nwCfg.CNIVersion)
+		if vererr != nil {
+			log.Printf("GetAsVersion failed with error %v", vererr)
+			plugin.Error(vererr)
+		}
+
+		if err == nil {
```
Comment (sharmasushant) [inline cni/network/network.go]: if err==nil && res!=nil

asking because if GetAsVersion fails, do we still want to print to stdout?

**Comment by sharmasushant** [comment]
@daschott we will get this merged by tomorrow.

**Comment by PatrickLang** [comment]
ping - is this still in the works?

**Comment by tamilmani1989** [comment]
@sharmasushant I addressed the review comment. Please approve it if everything is good.

**Pull request merged**